plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -1040,6 +1069,9 @@ def _identify_wanted_versions(self, package_info_generator):
             for unit, info in units.itervalues():
                 ret[unit] = info
 
+        if package_type == ids.TYPE_ID_RPM:
```

Comment:
I've been thinking about how this is an awkward pattern, and I recognize that the flow of code through this whole area really doesn't make it easy to do what you're trying to do. Doing the count comparison in the ``update_content()`` method doesn't seem any more appropriate than just doing it here. So what if instead of saving the count in this dictionary, you just did a comparison right here, and raise an exception right here?